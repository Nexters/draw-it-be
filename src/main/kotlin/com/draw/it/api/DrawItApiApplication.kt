package com.draw.it.api

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.ConfigurationPropertiesScan
import org.springframework.boot.runApplication
import org.springframework.scheduling.annotation.EnableScheduling

@SpringBootApplication
@ConfigurationPropertiesScan(basePackages = ["com.draw.it.api"])
@EnableScheduling
class DrawItApiApplication

fun main(args: Array<String>) {
	runApplication<DrawItApiApplication>(*args)
}
