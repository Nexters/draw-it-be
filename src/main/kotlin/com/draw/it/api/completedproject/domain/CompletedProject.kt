package com.draw.it.api.completedproject.domain

import com.draw.it.api.common.entity.BaseEntity
import jakarta.persistence.*
import org.hibernate.annotations.Comment

@Comment("Completed project")
@Entity
@Table(name = "completed_projects")
class CompletedProject(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    val id: Long? = null,

    @Comment("프로젝트 ID")
    @Column(name = "project_id", nullable = false)
    val projectId: Long,

    @Comment("이미지 URL")
    @Column(name = "image_url", nullable = false)
    val imageUrl: String
) : BaseEntity() {
    companion object {
        fun create(
            projectId: Long,
            imageUrl: String
        ): CompletedProject {
            return CompletedProject(
                projectId = projectId,
                imageUrl = imageUrl
            )
        }
    }
}
