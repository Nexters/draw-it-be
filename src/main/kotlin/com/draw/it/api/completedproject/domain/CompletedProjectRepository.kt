package com.draw.it.api.completedproject.domain

interface CompletedProjectRepository {
    fun save(completedProject: CompletedProject): CompletedProject
    fun findById(id: Long): CompletedProject?
    fun findByUserId(userId: Long): List<CompletedProject>
    fun findWithProjectNameByUserId(userId: Long): List<CompletedProjectWithName>
    fun deleteByProjectId(projectId: Long)
}