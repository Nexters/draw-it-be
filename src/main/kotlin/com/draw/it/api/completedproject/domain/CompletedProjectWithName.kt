package com.draw.it.api.completedproject.domain

import java.time.LocalDateTime

data class CompletedProjectWithName(
    val projectId: Long,
    val imageUrl: String,
    val projectName: String,
    val createdAt: LocalDateTime
)