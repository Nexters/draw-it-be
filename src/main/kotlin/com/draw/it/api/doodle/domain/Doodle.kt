package com.draw.it.api.doodle.domain

import com.draw.it.api.common.entity.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.Comment
import org.hibernate.annotations.SQLDelete
import org.hibernate.annotations.SQLRestriction

@Comment("두들 정보 테이블")
@Entity
@Table(name = "doodles")
@SQLDelete(sql = "UPDATE doodles SET is_deleted = true WHERE id = ?")
@SQLRestriction("is_deleted = false")
class Doodle(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    val id: Long? = null,

    @Comment("프로젝트 ID")
    @Column(name = "project_id", nullable = false)
    val projectId: Long,

    @Comment("두들 작성자 닉네임")
    @Column(name = "nickname", nullable = false)
    val nickname: String,

    @Comment("두들 편지 내용")
    @Column(name = "letter", columnDefinition = "TEXT")
    val letter: String?,

    @Comment("두들 이미지 URL")
    @Column(name = "image_url", nullable = false)
    val imageUrl: String,

    @Comment("사용자 확인 여부")
    @Column(name = "is_new_doodle_confirmed", nullable = false)
    val isNewDoodleConfirmed: Boolean = false,

    @Comment("두들 삭제 여부")
    @Column(name = "is_deleted", nullable = false, columnDefinition = "BOOLEAN DEFAULT FALSE")
    var isDeleted: Boolean = false,
) : BaseEntity() {

    fun delete() {
        this.isDeleted = true
    }
}
