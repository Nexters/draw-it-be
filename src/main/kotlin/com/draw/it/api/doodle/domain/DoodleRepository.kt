package com.draw.it.api.doodle.domain

interface DoodleRepository {
    fun save(doodle: Doodle): Doodle
    fun findById(id: Long): Doodle?
    fun findByProjectId(projectId: Long): List<Doodle>
}