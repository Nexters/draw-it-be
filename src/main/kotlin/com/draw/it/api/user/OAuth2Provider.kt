package com.draw.it.api.user

enum class OAuth2Provider {
    KAKAO,
    FACEBOOK
}