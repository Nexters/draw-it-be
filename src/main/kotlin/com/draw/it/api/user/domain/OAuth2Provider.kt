package com.draw.it.api.user.domain

enum class OAuth2Provider {
    KAKAO,
    FACEBOOK
}