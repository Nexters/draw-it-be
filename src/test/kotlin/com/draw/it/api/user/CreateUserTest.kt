package com.draw.it.api.user

import com.draw.it.api.user.domain.OAuth2Provider
import com.draw.it.common.IntegrationTest
import org.approvaltests.Approvals
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.jdbc.Sql
import kotlin.test.assertEquals

@IntegrationTest
@Sql("/user.sql")
class CreateUserTest {
    @Autowired
    private lateinit var createUser: CreateUser
    @Autowired
    private lateinit var getUser: ReadUser

    @Test
    fun `새로운 사용자를 생성하고 저장한다`() {
        val createdUserId = createUser.getOrCreateUser(
            name = "홍길동",
            provider = OAuth2Provider.KAKAO,
            providerId = "12345"
        )

        val savedUser = getUser.getUserById(createdUserId)
        Approvals.verify(savedUser)
    }

    @Test
    fun `기존 사용자가 있으면 기존 사용자 ID를 반환한다`() {
        // given
        val name = "홍길동"
        val provider = OAuth2Provider.KAKAO
        val providerId = "12345"

        // when
        val firstResult = createUser.getOrCreateUser(name, provider, providerId)
        val secondResult = createUser.getOrCreateUser(name, provider, providerId)

        // then
        assertEquals(firstResult, secondResult)
    }
}