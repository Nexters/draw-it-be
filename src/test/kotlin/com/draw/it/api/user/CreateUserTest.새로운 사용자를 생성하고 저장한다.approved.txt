UserResponse(id=1, name=홍길동, birthDate=null, provider=KAKAO, needsTutorial=true)
